upstream-sha256: ae082ae259484abe3d1c4d29367b3758640c6b2b83fb10db1fa4274073cde697
actual-sha256: 16f64d03984d28f7658a65c1420195d51a847658e07bc9453e312b0f6d5e04d6
--- upstream.expected
+++ tolk.actual
@@ -51,31 +51,33 @@
   50 Ops.AFTER           : 1          dtypes.float                             [23, 49]                         None
   51 Ops.INDEX           : 1          dtypes.float                             [50, 7]                          None
   52 Ops.LOAD            : 1          dtypes.float                             [51]                             None
-  53 Ops.CMPLT           : 1          dtypes.bool                              [52, 10]                         None
-  54 Ops.WHERE           : 1          dtypes.float                             [53, 10, 52]                     None
-  55 Ops.ADD             : 1          dtypes.float                             [22, 54]                         None
-  56 Ops.STORE           : 1          dtypes.void                              [21, 55]                         None
-  57 Ops.END             :            dtypes.void                              [56, 19]                         None
-  58 Ops.RANGE           : 103        dtypes.int                               [13, 57]                         (103, AxisType.REDUCE)
-  59 Ops.AFTER           : 103        dtypes.float                             [5, 11, 58]                      None
-  60 Ops.INDEX           : 103        dtypes.float                             [59, 7]                          None
-  61 Ops.LOAD            : 103        dtypes.float                             [60]                             None
-  62 Ops.ALLOC           :            dtypes.float                             []                               ParamArg(3, dtypes.float, 16, addrspace=AddrSpace.LOCAL)
-  63 Ops.INDEX           :            dtypes.float                             [62, 32]                         None
-  64 Ops.AFTER           :            dtypes.float                             [14, 57]                         None
-  65 Ops.INDEX           :            dtypes.float                             [64, 7]                          None
-  66 Ops.LOAD            :            dtypes.float                             [65]                             None
-  67 Ops.STORE           :            dtypes.void                              [63, 66]                         None
-  68 Ops.BARRIER         :            dtypes.void                              [67]                             None
-  69 Ops.AFTER           :            dtypes.float                             [62, 68]                         None
-  70 Ops.INDEX           : 103        dtypes.float                             [69, 58]                         None
-  71 Ops.LOAD            : 103        dtypes.float                             [70]                             None
-  72 Ops.ADD             : 103        dtypes.float                             [61, 71]                         None
-  73 Ops.STORE           : 103        dtypes.void                              [60, 72]                         None
-  74 Ops.END             :            dtypes.void                              [73, 58]                         None
-  75 Ops.AFTER           :            dtypes.float                             [5, 74]                          None
-  76 Ops.INDEX           :            dtypes.float                             [75, 7]                          None
-  77 Ops.LOAD            :            dtypes.float                             [76]                             None
-  78 Ops.CMPEQ           :            dtypes.bool                              [32, 7]                          None
-  79 Ops.STORE           :            dtypes.void                              [4, 77, 78]                      None
-  80 Ops.SINK            :            dtypes.void                              [79]                             KernelInfo(name='r_32_16_32_2', applied_opts=(Opt(op=OptOps.SPLIT, axis=2, arg=(16, AxisType.LOCAL, True)),), opts_to_apply=None, estimates=None, beam=0)
+  53 Ops.CMPLT           : 1          dtypes.bool                              [10, 52]                         None
+  54 Ops.CMPNE           : 1          dtypes.bool                              [52, 52]                         None
+  55 Ops.WHERE           : 1          dtypes.float                             [54, 52, 10]                     None
+  56 Ops.WHERE           : 1          dtypes.float                             [53, 52, 55]                     None
+  57 Ops.ADD             : 1          dtypes.float                             [22, 56]                         None
+  58 Ops.STORE           : 1          dtypes.void                              [21, 57]                         None
+  59 Ops.END             :            dtypes.void                              [58, 19]                         None
+  60 Ops.RANGE           : 103        dtypes.int                               [13, 59]                         (103, AxisType.REDUCE)
+  61 Ops.AFTER           : 103        dtypes.float                             [5, 11, 60]                      None
+  62 Ops.INDEX           : 103        dtypes.float                             [61, 7]                          None
+  63 Ops.LOAD            : 103        dtypes.float                             [62]                             None
+  64 Ops.ALLOC           :            dtypes.float                             []                               ParamArg(3, dtypes.float, 16, addrspace=AddrSpace.LOCAL)
+  65 Ops.INDEX           :            dtypes.float                             [64, 32]                         None
+  66 Ops.AFTER           :            dtypes.float                             [14, 59]                         None
+  67 Ops.INDEX           :            dtypes.float                             [66, 7]                          None
+  68 Ops.LOAD            :            dtypes.float                             [67]                             None
+  69 Ops.STORE           :            dtypes.void                              [65, 68]                         None
+  70 Ops.BARRIER         :            dtypes.void                              [69]                             None
+  71 Ops.AFTER           :            dtypes.float                             [64, 70]                         None
+  72 Ops.INDEX           : 103        dtypes.float                             [71, 60]                         None
+  73 Ops.LOAD            : 103        dtypes.float                             [72]                             None
+  74 Ops.ADD             : 103        dtypes.float                             [63, 73]                         None
+  75 Ops.STORE           : 103        dtypes.void                              [62, 74]                         None
+  76 Ops.END             :            dtypes.void                              [75, 60]                         None
+  77 Ops.AFTER           :            dtypes.float                             [5, 76]                          None
+  78 Ops.INDEX           :            dtypes.float                             [77, 7]                          None
+  79 Ops.LOAD            :            dtypes.float                             [78]                             None
+  80 Ops.CMPEQ           :            dtypes.bool                              [32, 7]                          None
+  81 Ops.STORE           :            dtypes.void                              [4, 79, 80]                      None
+  82 Ops.SINK            :            dtypes.void                              [81]                             KernelInfo(name='r_32_16_32_2', applied_opts=(Opt(op=OptOps.SPLIT, axis=2, arg=(16, AxisType.LOCAL, True)),), opts_to_apply=None, estimates=None, beam=0)
